### PR TITLE
Delete button

### PR DESCRIPTION
**Pull request by ErikOstien**
This adds a delete button to both vendor and product submission edit forms.
It does not add a delete button to the new forms.
We updated the models to support deletion.
We updated the admin controllers to have a destroy action.
We updated the views to have the button on the forms.
*And we made it so the admin facing tags index page does not route when tag is clicked on

### CODE REVIEW

File: app/views/admin/products/_form.html.haml [hunk 1]
```diff
@@ -99,4 +99,6 @@
 
     .form_group#submit_div
       = f.submit class: 'btn btn-success'
-      = button_tag 'Cancel', type: :button, onclick: 'confirmProductCancel()', id: :cancel_button, class: 'btn btn-danger'
\ No newline at end of file
+      -if @product.persisted?
```
Comment (jlabowitz) [inline app/views/admin/products/_form.html.haml:102]: What is the purpose of this call to persisted? I'm sure its necessary, but I'm unclear why.

File: app/views/admin/products/_form.html.haml [hunk 2]
```diff
@@ -99,4 +99,6 @@
...
+        = link_to 'Delete', admin_product_path, {method: :delete, data: {confirm: 'Are you sure you want to delete this product?'}, class: 'btn btn-danger', id: :delete_button}
```
Comment (jlabowitz) [inline app/views/admin/products/_form.html.haml:103]: I'm confused, I thought the browser doesn't understand methods other than post or get? How does it know what to do with delete?

File: app/views/application/_singleSymbolItems.html.haml [hunk 3]
```diff
@@ -8,6 +8,10 @@
           - path = {controller: "admin/#{myclass.name.downcase}s", action: "edit", id: item.id}
         - else
           - path = {controller: "#{myclass.name.downcase}s", action: "show", id: item.id}
-        = button_to "#{item.name}", path, {method: :get, class: 'prod-link'}
+        -if "#{myclass.name.downcase}" == "tag"
```
Comment (jlabowitz) [inline app/views/application/_singleSymbolItems.html.haml:11]: The changes to this file are unrelated to the pull request right? They just ensure that tags aren't clickable?

File: app/views/admin/products/_form.html.haml (hunk 1 above)
Comment (ErikOstien) [reply app/views/admin/products/_form.html.haml:102]: This checks to see if the product/vendor is already created or not. If it is, then we want to show the delete button, if not, then the cancel button suffices on a create new form.

File: app/views/admin/products/_form.html.haml (hunk 2 above)
Comment (ErikOstien) [reply app/views/admin/products/_form.html.haml:103]: This is using the HTML DELETE request

File: app/views/application/_singleSymbolItems.html.haml (hunk 3 above)
Comment (ErikOstien) [reply app/views/application/_singleSymbolItems.html.haml:11]: They are slightly unrelated, but it is a cleanup of the forms, and thus I added it in. It just makes sure we don't open the tags edit form page because it does not exist.

**Review by jlabowitz — APPROVED** [review]
All changes make sense to me now.

File: app/controllers/admin/products_controller.rb [hunk 4]
```diff
@@ -75,4 +75,14 @@ def update
       product_fail("update", product)
     end
   end
+
+  def destroy
```
Comment (nrleuenb) [inline app/controllers/admin/products_controller.rb:79]: Is there a way we can refactor out these destroy methods to be generic? The code is virtually the same.

**Review by nrleuenb — APPROVED** [review]
Left one comment on refactoring; otherwise LGTM.

File: app/controllers/admin/products_controller.rb (hunk 4 above)
Comment (dwang733) [reply app/controllers/admin/products_controller.rb:79]: Currently, the create and update also have the same problem. I think if we did refactor this, it should be a separate branch so that we can handle all 3 of the methods at the same time.

Comment (nrleuenb) [reply app/controllers/admin/products_controller.rb:79]: That sounds like the most elegant solution. Do we have time though?

Comment (dwang733) [reply app/controllers/admin/products_controller.rb:79]: Not sure tbh. CodeClimate isn't complaining for now, but it would be nice to have.

Comment (jlabowitz) [reply app/controllers/admin/products_controller.rb:79]: I think we should leave this pull request as is and potentially change this in the future if we have time. There are other more pertinent things, and as David said, CodeClimate isn't currently complaining.

Comment (nrleuenb) [reply app/controllers/admin/products_controller.rb:79]: If CodeClimate isn't complaining, then I'm not going to worry about it. As you said, it's a nice to have, but not urgent
